ocs_ci/deployment/mce.py

```diff
@@ -0,0 +1,243 @@
+            logger.info(f"Supported versions: {supported_versions}")
+
+        if not get_running_ocp_version() in supported_versions:
+            self.create_image_override()
+
+    def create_image_override(self):
+        """
+        Create hypershift image override cm
+        """
+        # Create image override configmap using the image override json
+        cmd = (
+            f"oc create cm {self.hypershift_override_image_cm} --from-file={constants.IMAGE_OVERRIDE_JSON}"
+            "-n {self.namespace}"
+        )
+        cmd_res = exec_cmd(cmd, shell=True)
+        assert cmd_res.returncode == 0
+
+        configmaps_obj = OCP(
+            kind=constants.CONFIGMAP,
+            namespace=constants.HYPERSHIFT_NAMESPACE,
+        )
+
+        assert configmaps_obj.is_exist(
+            resource_name=self.hypershift_override_image_cm
+        ), f"Configmap {self.hypershift_override_image_cm} does not exist in hypershift namespace"
+
+        # annotate multicluster engine operator with the override cm
+        self.multicluster_engine.annotate(
+            annotation=f"imageOverridesCM={self.hypershift_override_image_cm}"
+        )
+        self.multicluster_engine.wait_until_running()
+
+    def deploy_mce(self, check_mce_deployed=False, check_mce_ready=False):
+        """
+        Installs mce enabling software emulation.
+
+        Args:
+            check_mce_deployed (bool): If True, check if mce is already deployed. If so, skip the deployment.
+            check_mce_ready (bool): If True, check if mce is ready. If so, skip the deployment.
+        """
+        if check_mce_deployed:
+            if self.mce_hyperconverged_installed():
+                logger.info("mce operator is already deployed, skipping the deployment")
+                return
+
+        if check_mce_ready:
+            if self.post_install_verification(raise_exception=False):
+                logger.info("mce operator ready, skipping the deployment")
+                return
+
+        logger.info("Installing mce")
+        # we create catsrc with nightly builds only if config.DEPLOYMENT does not have mce_latest_stable
+        if not config.DEPLOYMENT.get("mce_latest_stable"):
+            # Create mce catalog source
+            self.create_mce_catalog_source()
+        # Create multicluster-engine namespace
+        self.create_mce_namespace()
+        # create mce subscription
+        self.create_mce_subscription()
+        # Deploy the multiclusterengine CR
+        self.create_multiclusterengine_operator()
+        # Check hypershift ns created
+        if not self.check_hypershift_namespace():
+            cmd = (
+                "oc patch mce multiclusterengine "
+                '-p \'{"spec":{"overrides":{"components":['
+                '{"enabled":true, "name":"hypershift"}'
+                "]}}}' --type=merge"
+            )
+            cmd_res = exec_cmd(cmd, shell=True)
+            assert cmd_res.returncode == 0
```

Comment:
same as above, assertion on this would not mean Product bug;
we have module with ocs-ci exceptions `ocs_ci.ocs.exceptions`, please use existing or add another with description